### PR TITLE
feat: allow passing googleAnalytics option

### PR DESCRIPTION
**Pull request by posva**
Allows passing `googleAnalytics: '<id>'` to the `themeConfig` option on the default theme.

I'm starting to think I should group this PR, #86 and another PR for Algolia search into a different vue theme. What do you think?

### CODE REVIEW

File: src/client/theme-default/index.ts [hunk 1]
```diff
@@ -6,10 +6,21 @@ import './styles/custom-blocks.css'
...
 
 const theme: Theme = {
   Layout,
-  NotFound
+  NotFound,
+  enhanceApp({ router, siteData }) {
+    const { googleAnalytics: GA_ID } = siteData.value.themeConfig
+    if (
+      GA_ID &&
+      process.env.NODE_ENV !== 'production' &&
```
Comment (yangmingshan) [inline src/client/theme-default/index.ts:18]: Did you mean `process.env.NODE_ENV === 'production'`?

Comment (posva) [inline src/client/theme-default/index.ts:18]:
```suggestion
      process.env.NODE_ENV === 'production' &&
```

Comment (posva) [reply src/client/theme-default/index.ts:18]: Yes, thanks

**Comment by kiaking** [comment]
I think having GA featurue is awesome! But since we've started Next development, let's do that there instead 👍

**Pull request closed without merging**